core/src/main/java/org/geysermc/geyser/translator/inventory/InventoryTranslator.java

```diff
@@ -453,35 +453,46 @@ public ItemStackResponse translateCraftingRequest(GeyserSession session, Invento
         ClickPlan plan = new ClickPlan(session, this, inventory);
         // Track all the crafting table slots to report back the contents of the slots after crafting
         IntSet affectedSlots = new IntOpenHashSet();
+        boolean reject = false;
```

Comment:
this doesn't seem necessary

Reply:
When you have a book request there will be multiple actions that come through. One needs to be handled properly, others will be rejected. If the ones to be rejected come first in the order and then have the rejection immediately returned, the early return cuts out the handling of the action that needs to be handled and it just errors. If we wait till all the actions are processed and only return when the proper one to handle shows up, it gets handled properly. It worked like this, with early returns it did not work.

Reply:
ah okay mb